src/hotspot/share/ci/ciMethodData.cpp

```diff
@@ -318,7 +314,7 @@ bool ciMethodData::load_data() {
 void ciReceiverTypeData::translate_receiver_data_from(const ProfileData* data) {
   for (uint row = 0; row < row_limit(); row++) {
     Klass* k = data->as_ReceiverTypeData()->receiver(row);
-    if (has_cld(k) && is_klass_loaded(k)) {
+    if (k != nullptr && k->is_loader_present_and_alive() && is_klass_loaded(k)) {
```

Comment:
This little part is actually incorrect. It should be just `k->class_loader_data() != nullptr` because there are different branch before for handling if `is_loader_alive()` and if not. I'll push the fix.